### PR TITLE
Fix errors en customer email field for registration form in FO and BO

### LINKED ISSUE #30475: Registration Form validation errors in email field
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Clean installation of PS 8.0.0.0 on PHP 8 or any PHP. When a new customer wants to register in frontend, normal email adresses work, like: name@domain.com. However: when emailadress is complicated like: part1-part2-part3part1-part2-part3part1-part2-part3@subdomain.domain.com , the form responds with 'invalid format'.  PS 1.7.8.7 (on PHP 7.x) did not have that problem.
Screenshot 2022-12-15 at 12 23 39

In backoffice, when creating a new customer the same way, i get error 'Unexpoected Error. [PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerException code 0]: Customer contains invalid field values'
Screenshot 2022-12-15 at 12 22 51

### Expected behavior

I expect that customer can register with dashes - before the @ sign and multiple dots . after the @ sign in the emailfield of the registry form, i.e. part1-part2-part3part1-part2-part3part1-part2-part3@subdomain.domain.com

### Steps to reproduce

On frontend of PS 8.0.0.0 on PHP 8:
1. go to Login/Register
2. fill in the fields accordingly, but use an email adress like: part1-part2-part3part1-part2-part3part1-part2-part3@subdomain.domain.com
3. click on Submit/Register

The frontend error will appear.

On backoffice:
1. go to Customers
2. click on 'Add New Customer'
3. fill in the fields accordingly, but use an email adress like: part1-part2-part3part1-part2-part3part1-part2-part3@subdomain.domain.com
4. click on Submit/Register

### PrestaShop version(s) where the bug happened

8.0.0

### PHP version(s) where the bug happened

8

### If your bug is related to a module, specify its name and its version

the PS registry form

### PR DESCRIPTION
**Pull request by lartist**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Fix errors en customer email field for registration form in FO and BO
| Type?             | bug fix
| Category?         | CO
| BC breaks?        |  no
| Deprecations?     |  no
| How to test?      | https://github.com/PrestaShop/PrestaShop/issues/30475
| Fixed ticket?     | Fixes #30475
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

**Review by kpodemski — COMMENTED** [review]
CS fixer is not happy.

**Review by kpodemski — DISMISSED** [review]
CI 🔴

**Comment by lartist** [comment]
> CI 🔴

Yeah I've seen. I'm trying to find a solution.
Actually here https://github.com/PrestaShop/PrestaShop/pull/31279/files#diff-3e424fffa7843b9b4fc56788de4feafa8246ff4a10f78e2920e319e835f08f5dL87 there are goods and bads :( .
I mean you can find rules which avoid email format without extension like john.doe@prestashop (good) but there are also rules like "length should not exceed 254 characters" which we don't want (bad). It would not be a good idea to spent a lot of time on refacto the legacy in front and I'm also limited with semver and cannot add a new Validation Class which would be a new feature :s

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 1]
```diff
@@ -191,6 +192,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     new NotBlank([
                         'message' => $this->trans('This field cannot be empty.', 'Admin.Notifications.Error'),
                     ]),
+                    new Length([
+                        'max' => DomainEmail::MAX_LENGTH,
+                        'maxMessage' => $this->trans(
+                            'This field cannot be longer than %limit% characters',
+                            'Admin.Notifications.Error',
+                            ['%limit%' => LastName::MAX_LENGTH]
```
Comment (FabienPapet) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                            ['%limit%' => DomainEmail::MAX_LENGTH]
```

**Review by FabienPapet — CHANGES_REQUESTED** [review]

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php (hunk 1 above)
Comment (lartist) [reply src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]: Done

**Review by FabienPapet — DISMISSED** [review]

File: src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php [hunk 2]
```diff
@@ -191,6 +192,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                     new NotBlank([
                         'message' => $this->trans('This field cannot be empty.', 'Admin.Notifications.Error'),
                     ]),
+                    new Length([
+                        'max' => DomainEmail::MAX_LENGTH,
+                        'maxMessage' => $this->trans(
+                            'This field cannot be longer than %limit% characters',
```
Comment (l-delin) [inline src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]:
```suggestion
                            'This field cannot be longer than %limit% characters.',
```

Comment (l-delin) [reply src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]: Just a small edit. Thanks!

Comment (lartist) [reply src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php]: Done @l-delin

**Comment by l-delin** [comment]
All good to go! Thanks :-)

**Review by FabienPapet — APPROVED** [review]

**Review by 0x346e3730 — APPROVED** [review]

File: classes/Validate.php [hunk 3]
```diff
@@ -80,11 +80,19 @@ public static function isEmail($email)
             return false;
         }
 
+        $validator = Validation::createValidator();
```
Comment (jolelievre) [inline classes/Validate.php:83]: Just a question, in the BO there is a constraint regarding the length, shouldn't there be the same in FO?
Or maybe there is but not handled in this method?

Comment (lartist) [reply classes/Validate.php:83]: Handle in the form

**Review by djoelleuch — APPROVED** [review]
Hello @lartist  ,

Automated tests --> OK
BO --> OK :heavy_check_mark: 

[bo.webm](https://user-images.githubusercontent.com/115152460/224343187-64ed35d5-d2af-48f6-afe7-6ae3397afe3c.webm)

FO --> OK :heavy_check_mark: 

[fo.webm](https://user-images.githubusercontent.com/115152460/224343276-ca21bc74-d147-46e8-a996-cbbf4cb951f5.webm)


The issue is well fixed , It's QA approved !
Thank you :rocket:

**Review by nicosomb — APPROVED** [review]

**Comment by jolelievre** [comment]
Thanks you @lartist and @djoelleuch

**Review by jolelievre — APPROVED** [review]

**Pull request merged**